lms/djangoapps/teams/csv.py

```diff
@@ -19,3 +28,222 @@ def load_team_membership_csv(course, response):
+    """
+    A manager class that is responsible the import process of csv file including validation and creation of
+    team_courseteam and teams_courseteammembership objects.
+    """
+
+    def __init__(self):
+        # the list of validation errors
+        self.error_list = []
+        self.teamset_names_list = []
+        # this is a dictionary of dictionaries that ensures that a student can belong to
+        # one and only one team in a teamset
+        self.teamset_membership_dictionary = {}
+        # dictionary that matches column index to a teamset name. Used when creating teams to get the right teamset
+        self.teamset_index_dictionary = {}
+        # the currently selected user
+        self.user = ''
+        self.number_of_record_added = 0
+        # stores the course module that will be used to get course metadata
+        self.course_module = ''
+        # stores the course for which we are populating teams
+        self.course = ''
+        self.max_erros = 0
+        self.max_erros_found = False
+
+    @property
+    def import_succeeded(self):
+        """
+        Helper wrapper that tells us the status of the import
+        """
+        return len(self.error_list) == 0
+
+    def set_team_membership_from_csv(self, course, input_file):
+        """
+        Assigns team membership based on the content of an uploaded CSV file.
+        Returns true if there were no issues.
+
+        Arguments:
+            course (CourseDescriptor): Course module for which team membership needs to be set.
+        """
+        self.error_list = []
+        self.teamset_names_list = []
+        self.teamset_membership_dictionary = {}
+        self.course = course
+        self.course_module = modulestore().get_course(course.id)
+        all_rows = [row for row in csv.reader(input_file.read().decode('utf-8').splitlines())]
+        header_row = all_rows[0]
+        if self.validate_teamsets(header_row):
+            # process student rows:
+            for i in range(1, len(all_rows)):
+                row_data = all_rows[i]
+                if row_data[0]:  # avoid processing rows with empty user names (excel copy and paste)
+                    user = self.reset_user(row_data[0])
+                    if user is None:
+                        continue
+                    if self.validate_user_entry(user, course) is False:
+                        row_data[0] = None
+                        continue
+
+                    row_data[0] = user
+
+                    if self.validate_user_to_team(row_data) is False:
+                        return False
+
+            if not self.error_list:
+                for i in range(1, len(all_rows)):
+                    self.add_user_to_team(all_rows[i], course)
+                return True
+            else:
+                return False
+        return False
+
+    def validate_teamsets(self, header_row):
+        """
+        Validates team set names. Returns true if there are no errors.
```

Comment:
Can you use the docstring to explain what validation is taking place? Otherwise I have to try to interpret the code to get what's going on, and ain't nobody got time for that.

Reply:
Also note that this method does two really different things:
1) it populates `user_ids_by_teamset_index`
2) it checks that the user-provided teamset names are valid teamset names according to the course's `teams_configuration`.
It'd be easier to understand if it were broken out into two separate methods.

Reply:
Docstring updated. I purposefully left it a little generic in case we need to add more.